File: app/src/main/java/com/jugger/tetris/Block/BlockTypes.java
package com.jugger.tetris.Block;


import android.graphics.Color;

import com.jugger.tetris.Basic.C;
import com.jugger.tetris.Basic.Pos;

public enum BlockTypes {

    I(Pos.create4NewPos(C.rows.getCount(),-1,2 , -1,3 , -1,4 , -1,5), Color.rgb(0,0,255)),
    L(Pos.create4NewPos(C.rows.getCount(),-1,2 , -1,3 , -1,4 , -2,2),Color.rgb(0,255,0)),
    J(Pos.create4NewPos(C.rows.getCount(),-1,2 , -2,2 , -2,3 , -2,4),Color.rgb(255,0,0)),
    O(Pos.create4NewPos(C.rows.getCount(),-1,3 , -1,4 , -2,3 , -2,4),Color.rgb(0,0,0)),
    S(Pos.create4NewPos(C.rows.getCount(),-1,4 , -1,5 , -2,3 , -2,4),Color.rgb(128,0,0)),
    T(Pos.create4NewPos(C.rows.getCount(),-1,3 , -2,2 , -2,3 , -2,4),Color.rgb(128,128,128)),
    Z(Pos.create4NewPos(C.rows.getCount(),-1,2 , -1,3 , -2,3 , -2,4),Color.rgb(128,128,0));

    private final Pos[] spawn;
    private final int c;
    private static int number = 7;
    public static BlockTypes[] array = new BlockTypes[]{I,J,L,O,S,T,Z};

    BlockTypes(Pos[] spawn, int c) {
        this.spawn = spawn;
        this.c = c;
    }

    public static BlockTypes getRandomBlockType() {
        int random = (int) (Math.random()*number);
        switch(random) {
            case 0:return I;
            case 1:return J;
            case 2:return L;
            case 3:return O;
            case 4:return S;
            case 5:return T;
            case 6:return Z;
            default: return I;
        }
    }


    public Pos[] getSpawnPos( ) {
        return spawn;
    }

    public int getColor( ) {
        return c;
    }
}

File: app/src/main/java/com/jugger/tetris/Block/BlockTypes.kt
package com.jugger.tetris.Block

import android.graphics.Color
import com.jugger.tetris.Basic.Point
import com.jugger.tetris.Constants

enum class BlockTypes(val spawnPos: Array<Point>, val color: Int) {
    I(Point.Companion.create4NewPos(Constants.rows.count, -1, 2, -1, 3, -1, 4, -1, 5), Color.rgb(0, 0, 255)), L(Point.Companion.create4NewPos(Constants.rows.count, -1, 2, -1, 3, -1, 4, -2, 2), Color.rgb(0, 255, 0)), J(Point.Companion.create4NewPos(Constants.rows.count, -1, 2, -2, 2, -2, 3, -2, 4), Color.rgb(255, 0, 0)), O(Point.Companion.create4NewPos(Constants.rows.count, -1, 3, -1, 4, -2, 3, -2, 4), Color.rgb(0, 0, 0)), S(Point.Companion.create4NewPos(Constants.rows.count, -1, 4, -1, 5, -2, 3, -2, 4), Color.rgb(128, 0, 0)), T(Point.Companion.create4NewPos(Constants.rows.count, -1, 3, -2, 2, -2, 3, -2, 4), Color.rgb(128, 128, 128)), Z(Point.Companion.create4NewPos(Constants.rows.count, -1, 2, -1, 3, -2, 3, -2, 4), Color.rgb(128, 128, 0));

    companion object {
        private const val number = 7
        var array = arrayOf(I, J, L, O, S, T, Z)
        val randomBlockType: BlockTypes
            get() {
                val random = (Math.random() * number).toInt()
                return when (random) {
                    0 -> I
                    1 -> J
                    2 -> L
                    3 -> O
                    4 -> S
                    5 -> T
                    6 -> Z
                    else -> I
                }
            }
    }

}